synapseutils/sync.py

```diff
@@ -796,14 +824,78 @@ def _get_file_entity_provenance_dict(syn, entity):
             raise  # unexpected error so we re-raise the exception
 
 
-def _write_manifest_data(filename, keys, data):
+def _convert_manifest_data_items_to_string_list(
+    items: typing.List[str],
+) -> typing.List[str]:
+    """
+    Handle coverting an individual key that contains a list of data into a list of strings
+    that can be written to the manifest file. This has specific logic around how to
+    handle datetime fields and non strings.
+
+    When working with datetime fields we are printing the ISO 8601 UTC representation of
+    the datetime.
+
+    When working with non strings we are printing the repr of the object.
+
+    Args:
+        items: The list of items to convert.
+
+    Returns:
+        The list of items converted to strings.
+    """
+    items_to_write = []
+    for item in items:
+        if isinstance(item, datetime.datetime):
+            items_to_write.append(
+                utils.datetime_to_iso(dt=item, include_milliseconds=False)
+            )
+        else:
+            items_to_write.append(item if isinstance(item, str) else repr(item))
+    return items_to_write
+
+
+def _convert_manifest_data_row_to_dict(row: dict, keys: typing.List[str]) -> dict:
+    """
+    Convert a row of data to a dict that can be written to a manifest file.
+
+    Args:
+        row: The row of data to convert.
+        keys: The keys of the manifest. Used to select the rows of data.
+
+    Returns:
+        The dict representation of the row.
+    """
+    data_to_write = {}
+    for key in keys:
+        data_for_key = row.get(key, "")
+        if isinstance(data_for_key, list):
+            items_to_write = _convert_manifest_data_items_to_string_list(data_for_key)
+            data_to_write[key] = ";".join(items_to_write)
```

Comment:
Yeah this is something I wasn't sure how we wanted to do it in the end, and I am open to whichever approach makes the most sense.

We already had an established pattern for the used and executed provenance fields: https://python-docs.synapse.org/explanations/manifest_tsv/#activityprovenance-fields

So that is the approach I took here.

Who would be a good source to make a decision on this direction?

Reply:
Let's float this on #synapse and tag milen and Andrew for feedback.

Reply:
I updated this to store values comma delimited instead of semi-colon delimited in https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1030/commits/f3c677daf9fac5ed6b50256b9a61e4c935effa34

In the end it'll look like:
![image](https://github.com/Sage-Bionetworks/synapsePythonClient/assets/17128019/04e08800-029b-413b-b1cb-9d599251a1b9)

`\` will be an escape character to allow a comma to be used in a string.